gtda/diagrams/representations.py

```diff
@@ -450,59 +448,67 @@ def plot(self, Xt, sample=0, homology_dimensions=None, plotly_params=None):
                 else:
                     homology_dimensions_arr = np.array(
                         self.homology_dimensions_)
-                    ix = np.flatnonzero(homology_dimensions_arr == dim)[0]
-                    _homology_dimensions.append((ix, dim))
-
+                    inv_idx = np.flatnonzero(homology_dimensions_arr == dim)[0]
+                    _homology_dimensions.append((inv_idx, dim))
+
+        layout_axes_common = {
+            "type": "linear",
+            "ticks": "outside",
+            "showline": True,
+            "zeroline": True,
+            "linewidth": 1,
+            "linecolor": "black",
+            "mirror": False,
+            "showexponent": "all",
+            "exponentformat": "e"
+            }
         layout = {
             "xaxis1": {
                 "side": "bottom",
-                "type": "linear",
-                "ticks": "outside",
                 "anchor": "y1",
-                "showline": True,
-                "zeroline": True,
-                "showexponent": "all",
-                "exponentformat": "e"
+                **layout_axes_common
                 },
             "yaxis1": {
                 "side": "left",
-                "type": "linear",
-                "ticks": "outside",
                 "anchor": "x1",
-                "showline": True,
-                "zeroline": True,
-                "showexponent": "all",
-                "exponentformat": "e"
+                **layout_axes_common
                 },
             "plot_bgcolor": "white",
-            "title": f"Landscape representation of diagram {sample}"
             }
 
         Xt_sample = Xt[sample]
-        for ix, dim in _homology_dimensions:
-            layout_dim = layout.copy()
-            layout_dim["title"] = "Persistence landscape for homology " + \
-                                  "dimension {}".format(int(dim))
-            fig = gobj.Figure(layout=layout_dim)
-            fig.update_xaxes(zeroline=True, linewidth=1, linecolor="black",
-                             mirror=False)
-            fig.update_yaxes(zeroline=True, linewidth=1, linecolor="black",
-                             mirror=False)
-
-            n_layers = Xt_sample.shape[1]
+        n_layers = Xt_sample.shape[1]
+        subplot_titles = [f"H{int(dim) if dim != np.inf else np.inf}"
+                          for _, dim in _homology_dimensions]
+        fig = make_subplots(rows=len(_homology_dimensions), cols=1,
+                            subplot_titles=subplot_titles)
+        many_homology_dim = len(_homology_dimensions) - 1
```

Comment:
Maybe something that starts with `is_` or `has_` to indicate that it's used as a boolean?

Reply:
Done.